src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -176,17 +178,18 @@ private static string GetArchitectureString()
         // platform specific file name of the extension library
         private static string GetNativeLibraryFilename()
         {
+            string architecture = GetArchitectureString();
```

Comment:
No, GetArchitectureString only checks the 32/64-bit aspect; the only platform checks in here are for OS. Did you mean in GetPlatformString? I'm sure there are other ways of doing this, but I was aiming for minimal changes in this PR... this approach works, and the overhead is tiny and one-off.